torch/csrc/jit/python_tracer.cpp

```diff
@@ -40,9 +40,19 @@ void initPythonTracerBindings(PyObject* module_) {
       ASSERT_UNEXPIRED("pop_scope");
       s.pop_scope();
     })
-    .def("export", [](TracingState& s, const std::vector<at::Tensor>& initializers, int64_t onnx_opset_version) {
+    .def("export", [](TracingState& s, const std::vector<at::Tensor>& initializers,
+                      int64_t onnx_opset_version, bool defer_weight_export=false) {
       ASSERT_UNEXPIRED("export");
-      return py::bytes(ExportGraph(s.graph, initializers, onnx_opset_version));
+      std::string graph;
+      RawDataExportMap export_map;
+      std::tie(graph, export_map) = ExportGraph(
+        s.graph, initializers, onnx_opset_version, defer_weight_export);
+      std::unordered_map<std::string, py::bytes> python_serialized_export_map;
+      for (auto& kv : export_map) {
+        python_serialized_export_map[kv.first] = py::bytes(kv.second);
```

Comment:
Isn't this doing another copy? I know, "don't prematurely optimize", but it seems to me that you are:

1. Doing a copy into a std::string for the export map
2. Doing another copy into the Python export map
3. Doing a final copy when you finally decide where the string should go

These are big strings, so every copy matters.

Reply:
I think I can remove the first copy by storing an `at::Tensor` rather than a raw buffer. As for 2 and 3, not sure I can get rid of those, since afaict `py::bytes` requires a copy. To clarify, is #3 in python?

Reply:
Yes, 3 is in Python.

Suppose you pass a tensor back into Python. In that case, no copies happen at all, and then you can blat the tensor data directly into the Python APIs you want to use.